pyproject.toml

```diff
@@ -508,6 +512,13 @@ ignore = [
     "RUF015",  # Prefer `next(...)` over single element slice
     "RUF021",  # Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear
     "RUF029",  # Function is declared `async`, but doesn't `await` or use `async` features.
+    "RUF036",  # `None` not at the end of the type annotation.
+    "RUF039",  # [*] First argument to `re.compile()` is not raw string
+    "RUF043",  # Pattern passed to `match=` contains metacharacters but is neither escaped nor raw
+    "RUF051",  # [*] Use `pop` instead of `key in dict` followed by `del dict[key]`
+    "RUF052",  # Local dummy variable `_meta` is accessed
+    "RUF056",  # [*] Avoid providing a falsy fallback to `dict.get()` in boolean test positions. The default fallback `None` is already falsy.
+    "RUF100",  # [*] Unused `noqa` directive (non-enabled: `INP001`)
```

Comment:
should we just removed INP001 from the code if it's not enabled instead of adding this exception ?

Reply:
I didn't want to make any code changes that weren't strictly required by this PR. For those with the `[*]` mark ruff should have auto-fix options and would be easy to do as follow up PRs.